### PR TITLE
feat(html/minifier): improve the `remove_redundant_attributes` option

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Due https://github.com/kangax/html-minifier/issues/1075

A lot of CSS frameworks use (and not only):
```
input[type=text] {
  /* styles */
}
```

So it is not safe to remove them by default, anyway I implemented the `safe` mode - it will remove:
- deprecated attributes (they doesn't used in CSS almost always, expect very old frameworks, which already doesn't supported)
- SVG attributes, it  is common ptractide don't use them in CSS
- attributes from metadata elements, because they will not be rendered

**BREAKING CHANGE:**

Yes

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_html_minifier/src/option.rs [hunk 1]
```diff
@@ -44,6 +44,20 @@ pub enum CollapseWhitespaces {
     OnlyMetadata,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "kebab-case")]
+pub enum RemoveRedundantAttributes {
```
Comment (kdy1) [inline crates/swc_html_minifier/src/option.rs:50]: Let's implement `Default` for this type instead of using `#[serde(default = "")]`.

Comment (kdy1) [reply crates/swc_html_minifier/src/option.rs:50]: You can then change `#[serde(default = "...")]` to `#[serde(default)]`

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Comment by alexander-akait** [comment]
@kdy1 Done, for another enum too :+1:

**Review by kdy1 — CHANGES_REQUESTED** [review]
CI faileda

**Comment by alexander-akait** [comment]
@kdy1 Fixed again :smile:

**Review by kdy1 — APPROVED** [review]

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_html_minifier --breaking

**Pull request merged**